pkg/subctl/operator/submarinerop/serviceaccount/ensure.go

```diff
@@ -84,10 +85,16 @@ func ensureServiceAccounts(kubeClient kubernetes.Interface, namespace string) (b
 		return false, errors.Wrap(err, "error provisioning globalnet ServiceAccount resource")
 	}
 
+	createdDiagnoseSA, err := serviceaccount.EnsureFromYAML(kubeClient, namespace,
```

Comment:
I'm not sure why this needs to be the responsibility of submariner-operator?

Since this a standalone capability, shouldn't the "diagnose operator" be the one ensuring this service account?

Reply:
We don't have any diagnose operator as diagnose doesn't need anything other than relevant RBAC for it. So added to `submariner-operator` as it was the best fit. Creating a whole new operator just to deploy RBAC seems to be overkill. Note that we do deploy globalnet SAs etc. even on deployments without globalnet, so we have a precedent.

Reply:
Ah from the original EP I thought that there's going to be an operator, because of:

> * A new `SubmarinerDiagnose` CRD will be added to SubmarinerAddon. OCM UI will create this CR with relevant values
>  to trigger running diagnose from OCM UI.`
> * A new `SubmarnerDiagnoseController` Controller will be added to OCM SubmarinerAddon. This controller will spawn the SubmarinerDiagnose

So if that all lives in OCM this part can just as well live there, and the proposed controller will make sure it has the correct RBAC.
I guess it's more relevant to that specific use case?

Also the fact that globalnet RBAC is deployed even without globalnet is sloppy and probably wrong? But that's not an excuse IMHO

Reply:
@mkolesnik This will also allow non OCM deployments to run diagnostics by just creating the Job. This enables use cases like creating a CronJob to run diagnostics periodically, run Diagnostics using golang instead shell scripts, run Diagnose as part of our E2E, run it as part of our operator deployment to make sure everything deployed correctly etc. Permissions required for diagnose are very closely tied to internals of diagnose. Having it in operator will help make sure we're updating RBAC when we add new features to diagnose too.

Agree on Globalnet RBAC, it is just bad security practice.

Reply:
I agree this can be useful for running diagnose by just creating the job, but I don't think this will be done much if at all.

Is `subctl diagnose` going to run the job and collect the result?

Also I'm still uneasy with just deploying these permissions even though they might never be used.
I would rather they be deployed on a need-for basis, e.g. we could have a flag in the `Submariner` CR indicating this capability is enabled and that would deploy these (or make sure they're deleted if changed from enabled to disabled).

Consider from a security standpoint a model of "less attack surface" in this context, the less permissions we're giving out needlessly, the more secure the system is.

Reply:
The new SA can’t be used unless you have the corresponding token; and if anyone is in a position to get the token, we’ve lost anyway.

I do think the roles currently grant permissions that aren’t needed for `diagnose`, it would be worth checking which ones can be removed (especially those granting the ability to make changes).